code/VintageEngineering/Transport/BlockPipeBase.cs

```diff
@@ -61,9 +61,9 @@ public override void OnNeighbourBlockChange(IWorldAccessor world, BlockPos pos,
         {
             // Redetect any potential connections as something changed.
             BEPipeBase pipebe = api.World.BlockAccessor.GetBlockEntity(pos) as BEPipeBase;
-            if (pipebe != null)
-            {                
-                pipebe.MarkPipeDirty(world, true);                 
+            if (pipebe != null) // pipebe exist but can be not initialize( so Api is not set ) IDK why or if it is a bug or not; maybe add a bool in BEPipeBase true when fully initialised
```

Comment:
Knowing how the game loads and processes information is vital to knowing how to tackle the problem. It's taken me years to learn this process and I still don't know half of what the game does.
A Block Entity can exist but not be initialized because the game goes through several loading steps.